File: leetcode-solutions/2106-find-greatest-common-divisor-of-array/solution.java
class Solution {
    private int gcd(int a, int b) {
        if (b == 0) return a;
        return gcd(b, a % b);
    }

    public int findGCD(int[] nums) {
        int min = Integer.MAX_VALUE, max = Integer.MIN_VALUE;
        for (int x : nums) {
            min = Math.min(min, x);
            max = Math.max(max, x);
        }
        return gcd(min, max);
    }
}

File: leetcode-solutions/2106-find-greatest-common-divisor-of-array/solution.cpp
#include <vector>
#include <algorithm>
#include <numeric> // For std::gcd

class Solution {
public:
    int findGCD(std::vector<int>& nums) {
        // Use std::minmax_element to find both min and max in a single pass
        auto [min_it, max_it] = std::minmax_element(nums.begin(), nums.end());
        
        // Return the greatest common divisor of the two values
        // std::gcd is available in <numeric> since C++17
        return std::gcd(*min_it, *max_it);
    }
};
